test/integration/scheduler/scheduler_test.go

```diff
@@ -1407,6 +1408,181 @@ var _ = ginkgo.Describe("Scheduler", func() {
+			gomega.Eventually(func() error {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: cohort.Name}, cohort)
+				cohort.Spec.ResourceGroups[0].Flavors[0].Resources[0].NominalQuota = resource.MustParse("10")
+				return k8sClient.Update(ctx, cohort)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+			expectAdmission := testing.MakeAdmission(cq.Name).Assignment(corev1.ResourceCPU, "on-demand", "10").Obj()
+			util.ExpectWorkloadToBeAdmittedAs(ctx, k8sClient, wl, expectAdmission)
+			util.ExpectPendingWorkloadsMetric(cq, 0, 0)
+			util.ExpectAdmittedWorkloadsTotalMetric(cq, 1)
+		})
+
+		ginkgo.It("Should not admit workload when cohort loses capacity", func() {
+			cq = testing.MakeClusterQueue("clusterqueue").
+				Cohort("cohort").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "0").Obj(),
+				).Obj()
+			queue := testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, queue)).Should(gomega.Succeed())
+
+			ginkgo.By("cohort created")
+			cohort = testing.MakeCohort("cohort").
+				ResourceGroup(*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "10").Obj()).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohort)).Should(gomega.Succeed())
+
+			ginkgo.By("cohort capacity reduced")
+			gomega.Eventually(func() error {
+				_ = k8sClient.Get(ctx, types.NamespacedName{Name: cohort.Name}, cohort)
+				cohort.Spec.ResourceGroups[0].Flavors[0].Resources[0].NominalQuota = resource.MustParse("9")
+				return k8sClient.Update(ctx, cohort)
+			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			ginkgo.By("workload not admitted")
+			wl = testing.MakeWorkload("wl", ns.Name).Queue(queue.Name).
+				Request(corev1.ResourceCPU, "10").Obj()
+			gomega.Expect(k8sClient.Create(ctx, wl)).Should(gomega.Succeed())
+			util.ExpectWorkloadsToBePending(ctx, k8sClient, wl)
+			util.ExpectAdmittedWorkloadsTotalMetric(cq, 0)
+		})
+
+		ginkgo.It("Should not admit workload when cohort deletion results in insufficient capacity", func() {
+			cq = testing.MakeClusterQueue("clusterqueue").
+				Cohort("cohort").
+				ResourceGroup(
+					*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "0").Obj(),
+				).Obj()
+			queue := testing.MakeLocalQueue("queue", ns.Name).ClusterQueue(cq.Name).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cq)).Should(gomega.Succeed())
+			gomega.Expect(k8sClient.Create(ctx, queue)).Should(gomega.Succeed())
+
+			ginkgo.By("cohort created")
+			cohort = testing.MakeCohort("cohort").
+				ResourceGroup(*testing.MakeFlavorQuotas("on-demand").Resource(corev1.ResourceCPU, "5").Obj()).Obj()
+			gomega.Expect(k8sClient.Create(ctx, cohort)).Should(gomega.Succeed())
+
+			ginkgo.By("cohort deleted")
+			gomega.Expect(util.DeleteObject(ctx, k8sClient, cohort)).To(gomega.Succeed())
+
+			ginkgo.By("workload not admitted")
```

Comment:
There is potential for flakiness here, as the deletion might not have been applied before you create this workload.

Reply:
Deleted.